core/dovecot/start.py

```diff
@@ -6,23 +6,40 @@
 import glob
 import multiprocessing
 import tenacity
+import logging as log
+import sys
 
 from tenacity import retry
 from podop import run_server
 
+log.basicConfig(stream=sys.stderr, level=os.environ.get("LOG_LEVEL", "WARNING"))
 
 def start_podop():
     os.setuid(8)
-    run_server(3 if "DEBUG" in os.environ else 0, "dovecot", "/tmp/podop.socket", [
+    run_server(0, "dovecot", "/tmp/podop.socket", [
 		("quota", "url", "http://admin/internal/dovecot/§"),
 		("auth", "url", "http://admin/internal/dovecot/§"),
 		("sieve", "url", "http://admin/internal/dovecot/§"),
     ])
 
-convert = lambda src, dst: open(dst, "w").write(jinja2.Template(open(src).read()).render(**os.environ))
+def convert(src, dst):
+    logger = log.getLogger("convert()")
+    logger.debug("Source: %s, Destination: %s", src, dst)
+    open(dst, "w").write(jinja2.Template(open(src).read()).render(**os.environ))
+
+@retry(
+    stop=tenacity.stop_after_attempt(100),
+    wait=tenacity.wait_random(min=2, max=5),
+    before=tenacity.before_log(log.getLogger("tenacity.retry"), log.DEBUG),
+    before_sleep=tenacity.before_sleep_log(log.getLogger("tenacity.retry"), log.INFO),
+    after=tenacity.after_log(log.getLogger("tenacity.retry"), log.DEBUG)
+    )
+def resolve(hostname):
+    logger = log.getLogger("resolve()")
+    logger.info(hostname)
```

Comment:
When looking at the logs from a more user-oriented perspective, and even if the user is an admin, a brief help on what is trying to get resolved here might help …. Something like: `logger.info("Trying to resolve name of required container: %s", hostname)`, maybe?

Reply:
The string passed to `log.getLogger()` is used as log entry tag. So output looks like:
```
imap_1       | INFO:resolve():front
```
So I know resolve was called for front. I added this as a prelude to tenacity, because it does not pass the function arguments (like host) in its log output. So when failing the logs look like:
```
imap_1       | DEBUG:tenacity.retry:Starting call to '__main__.resolve', this is the 1st time calling it.
imap_1       | INFO:resolve():front
imap_1       | DEBUG:tenacity.retry:Finished call to '__main__.resolve' after 0.036(s), this was the 1st time calling it.
imap_1       | INFO:tenacity.retry:Retrying __main__.resolve in 4.1597329924216915 seconds as it raised [Errno -2] Name does not resolve.
imap_1       | DEBUG:tenacity.retry:Starting call to '__main__.resolve', this is the 2nd time calling it.
```

However, I've nothing against more verbosity if someone wants to implement such things.